pkg/controller/operators/openshift/helpers_test.go

```diff
@@ -441,6 +441,70 @@ func TestIncompatibleOperators(t *testing.T) {
 				incompatible: nil,
 			},
 		},
+		{
+			description: "ClusterZ",
+			cv: configv1.ClusterVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "version",
+				},
+				Status: configv1.ClusterVersionStatus{
+					Desired: configv1.Update{
+						Version: "1.0.1", // Next Y-stream is 1.1.0, NOT 1.1.1
+					},
+				},
+			},
+			in: skews{
+				{
+					name:                "almond",
+					namespace:           "default",
+					maxOpenShiftVersion: "1.1.2",
+				},
+				{
+					name:                "beech",
+					namespace:           "default",
+					maxOpenShiftVersion: "1.1",
+				},
+			},
+			expect: expect{
+				err:          false,
+				incompatible: nil,
+			},
+		},
+		{
+			description: "ClusterPre",
+			cv: configv1.ClusterVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "version",
+				},
+				Status: configv1.ClusterVersionStatus{
+					Desired: configv1.Update{
+						Version: "1.1.0-pre", // Next Y-stream is 1.1.0, NOT 1.2.0
```

Comment:
Do we need a test for `<major>.<minor>.<patch>-<pre-release>+<build>` Would we ever need to specify that kind of format downstream?

Reply:
I omitted it because build versions have no partial order in semver, but it probably best to not make any assumptions when testing -- I'll add another test case